src/housing_unit_inventory/calculate.py

```diff
@@ -0,0 +1,107 @@
+# import logging
+import warnings
+from datetime import datetime
+
+# import numpy as np
+# import pandas as pd
+
+
+def update_unit_count(parcels_df):
+    """Update unit counts in-place for single family, duplex, and tri/quad
+
+    Args:
+        parcels_df (pd.DataFrame): The evaluated parcel dataset with UNIT_COUNT, HOUSE_CNT, SUBTYPE, and NOTE columns
+    """
+
+    #: FIXME: Field names are good, but subtype values may be davis-specific. Do we need these checks, or can we just
+    #: use the address points instead?
+
+    # fix single family (non-pud)
+    zero_or_null_unit_counts = (parcels_df['UNIT_COUNT'] == 0) | (parcels_df['UNIT_COUNT'].isna())
+    parcels_df.loc[(zero_or_null_unit_counts) & (parcels_df['SUBTYPE'] == 'single_family'), 'UNIT_COUNT'] = 1
```

Comment:
**question**: Is there a reason why you are using bitwise operators rather than the `or`/`and` boolean operators? I always thought that bitwise was for bits or numbers but I could be wrong. I think that the outcome would be similar, if not exactly the same, for this code. I'm just wondering if I'm missing something.

Reply:
Within a pandas `.loc` statement, you have to use `| &` instead of or/and. The first line there was me pulling a complex statement out into its own variable... not sure if it makes a difference now that it's outside of a `.loc`...